pkg/chunked/compression_linux.go

```diff
@@ -254,14 +258,20 @@ func readZstdChunkedManifest(blobStream ImageSourceSeekable, tocDigest digest.Di
 		if err != nil {
 			return nil, nil, nil, 0, err
 		}
-		decodedTarSplit, err = decodeAndValidateBlob(tarSplit, tarSplitLengthUncompressed, toc.TarSplitDigest.String())
+		fd, err := unix.Open(tmpDir, unix.O_TMPFILE|unix.O_RDWR|unix.O_CLOEXEC, 0o600)
```

Comment:
just a drive by comment
The open man page says O_TMPFILE needs support from the file system. Is that a real problem we would need to care about? Could this affect any users on an uncommon fs that does not have support for it?

Reply:
good point.  I've added a fallback when the underlying file system doesn't support it and adapted the other user of `O_TMPFILE`

Reply:
Is there really no well-known Go library out there wrapping this? In my Rust projects we use https://docs.rs/cap-tempfile/latest/cap_tempfile/struct.TempFile.html and I've contributed to the crate.